src/Dialog.cc

```diff
@@ -75,3 +80,141 @@ QQuickItem *Dialog::RootItem() const
   return dialogItem;
 }
 
+/////////////////////////////////////////////////
+bool Dialog::UpdateConfigAttribute(const std::string &_path,
+  const std::string &_attribute, const bool _value) const
+{
+  if (_path.empty())
+  {
+    ignerr << "Missing config file" << std::endl;
+    return false;
+  }
+
+  // Use tinyxml to read config
+  tinyxml2::XMLDocument doc;
+  auto success = !doc.LoadFile(_path.c_str());
+  if (!success)
+  {
+    ignerr << "Failed to load file [" << _path << "]: XMLError"
+            << std::endl;
+    return false;
+  }
+
+  // Update attribute value for the correct dialog
+  for (auto dialogElem = doc.FirstChildElement("dialog");
+    dialogElem != nullptr;
+    dialogElem = dialogElem->NextSiblingElement("dialog"))
+  {
+    if(dialogElem->Attribute("name") == this->objectName().toStdString())
+    {
+      dialogElem->SetAttribute(_attribute.c_str(), _value);
+    }
+  }
+
+  // Write config file
+  tinyxml2::XMLPrinter printer;
+  doc.Print(&printer);
+
+  std::string config = printer.CStr();
+  std::ofstream out(_path.c_str(), std::ios::out);
+  if (!out)
+  {
+    ignerr << "Unable to open file: " << _path
+           << ".\nCheck file permissions.\n";
+  }
+  else
+    out << config;
+
+  return true;
+}
+
+/////////////////////////////////////////////////
+void Dialog::SetDefaultConfig(const std::string &_config)
+{
+  this->dataPtr->config = _config;
+}
+
+/////////////////////////////////////////////////
+std::string Dialog::ReadConfigAttribute(const std::string &_path,
+  const std::string &_attribute) const
+{
+  tinyxml2::XMLDocument doc;
+  std::string value {""};
+  std::string config = "<?xml version=\"1.0\"?>\n\n";
+  tinyxml2::XMLPrinter defaultPrinter;
+  bool configExists{true};
+  std::string dialogName = this->objectName().toStdString();
+
+  auto Value = [&_attribute, &doc, &dialogName]()
+  {
+    // Process each dialog
+    // If multiple attributes share the same name, return the last one
```

Comment:
It looks like the new logic actually returns the first one, right?

Reply:
copy pasting comments without re-reading :upside_down_face: 
thanks for all the catches!